ffi/src/lib.rs

```diff
@@ -39,28 +38,42 @@ pub unsafe trait Handle {
 ///
 /// Type implementing the trait must be a robust type with a guaranteed C ABI. Care must be taken
 /// not to dereference pointers whose referents don't implement `ReprC`; they are considered opaque
-pub unsafe trait ReprC: Sized {}
+pub unsafe trait ReprC: Copy + Sized {}
 
-/// Used to do a cheap reference-to-[`ReprC`]-reference conversion
-pub trait AsReprCRef<'itm> {
-    /// Robust C ABI compliant representation of &[`Self`]
-    type Target: ReprC + 'itm;
+/// Indicates that the type can be converted into an FFI compatible representation
+pub trait FfiType: Sized {
+    /// Corresponding FFI-compatible type
+    type ReprC: ReprC;
+}
 
-    /// Convert from &[`Self`] into [`Self::Target`].
-    fn as_ref(&'itm self) -> Self::Target;
+/// Type that can be returned from an FFI function via out-pointer function argument
+pub trait Output: FfiType {
```

Comment:
This trait is tricky, and might be too permissive, I should add tests as a guard